### PR TITLE
Updated install function

### PR DESCRIPTION
**Pull request by delvedor**
I have updated the `install` function removing the `update` parameter, because the new UI manages in a different way the refresh after the installation and because in my opinion `install` should only do his job, everything else can be done in the callback.

I also updated the callback inside `check-node.js`, because was passing the error as second argument.

### CODE REVIEW

File: lib/check-node.js [hunk 1]
```diff
@@ -3,10 +3,10 @@ const semver = require('semver')
 
 module.exports = (cb) => {
   exec('node --version', (error, stdout, stderr) => {
-    if (error) return cb(error)
+    if (error) return cb(error, null)
     const output = (stdout + stderr).replace(/(\r\n|\n|\r| )/gm, '')
     const sem = semver.valid(output)
     if (sem) return cb(null, sem)
-    cb(null, new Error(output))
+    cb(new Error(output), null)
```
Comment (addaleax) [inline lib/check-node.js:10]: The changes here are unrelated to the rest, right? I’d suggest separating that out into a different commit/different PR, then.

Comment (delvedor) [reply lib/check-node.js:10]: I thought about doing it in a different commit, but at the end I made all in one because the change is related to the install function, I wanted to standardize the callback in every part of the "pipeline" with `(error, version)`.
`browser.js -> install.js -> check-node.js -> browser.js`

I hope I made myself clear :)

**Comment by delvedor** [comment]
@mikeal @addaleax @JoeDoyle23 I think this can be merged, is just a little fix to make the install function compliant with  the UI code :)

**Comment by addaleax** [comment]
Mh, one reason why I suggested that the changes to `lib/check-node.js` be in a different commit/different PR would be that that looks like an obviously correct and logically independent bugfix, and I would feel pretty comfortable merging that rather swiftly.

The rest of the changes… not sure, maybe I lack a global look at things (and I haven’t tried the current `master` of the installer myself, will do when I have the time). It doesn’t really look useless to me to have a way of notifying the UI of the “Download complete” situation.

**Comment by delvedor** [comment]
Ok, you are right about the different commit for `lib/check-node.js`, I'll update it.

Regarding the update function at the moment during download/installation the UI does not change, it has a [spinner](https://cloud.githubusercontent.com/assets/4865608/18231338/94acb902-72b7-11e6-94e0-e16c82f96fff.png) and waits for the callback of success/[fail](https://cloud.githubusercontent.com/assets/4865608/18507538/15428036-7a71-11e6-9901-214f8b7ad193.png) of the entire process. That's why I'm proposing to remove that function.
If you feel that the `'Download Finished'` must be notified in the UI we can discuss about it :)

**Comment by addaleax** [comment]
> If you feel that the `'Download Finished'` must be notified in the UI we can discuss about it :)

Not necessarily. I find progress bars / status indicators for things like downloads extremely useful sometimes, especially when I’m on a slow connection.

I see no need to block this PR on that, but maybe something like an `EventEmitter` for notifying the UI would give some flexibility in that direction without forcing the UI to display information in a specific way.

(sorry for not following up sooner!)

**Review by addaleax — APPROVED** [review]

**Comment by delvedor** [comment]
I like the idea of the event emitter could be even simpler to implement :)

**Comment by addaleax** [comment]
@delvedor If you want to, feel free to go ahead and implement that. On a more general note, it might be worthwhile to think about separating the installer logic from the UI and turning the former into its own package?

**Comment by delvedor** [comment]
I'll definitely continue to develop this :)

Regarding the separate package for the installer I agree with you, put it in a separate package could be a good idea.
